core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -333,6 +333,8 @@ private void commitCreateTransaction() {
       // the commit failed and no files were committed. clean up each update
       if (!ops.requireStrictCleanup() || e instanceof CleanableFailure) {
         cleanAllUpdates();
+      } else if (ops.requireStrictCleanup()) {
```

Comment:
I believe this comment needs rewording: 
https://github.com/apache/iceberg/blob/afc30818b734879e53a24f90f034685cf8fc56bc/core/src/main/java/org/apache/iceberg/TableOperations.java#L120

"would it now say if requiresStrictCleanup is true, we would NOT cleanup if it was not a cleanableFailure", is there a easier way to represent this if-else condition in that case."


+1 to being conservative we have been bitten multiple times with premature cleanup by spark because the code raised unchecked exceptions such as OutOfMemoryError. On that note, how are unchecked exceptions that derive from Error class handled from cleanup perspective.

Reply:
No longer applicable. Just updated the approach; it's still conservative but it puts the handling more on the Spark integration side to explicitly handle CleanableFailure rather than the original approach which attempted to normalize the unknown cases to CommitStateUnknownException; I think it's better because that "normalization" may be quite hard and there maybe other cases.